### ISSUE TITLE
VM for quick development environment deployment

### ISSUE THREAD

**Issue by vaskas** [BerryFarm/berrymuch#21]
I want to have a
* a base box
* a provisioning script
So that anyone can roll their own BB10 development environment out quickly

**Comment by vaskas** [comment]
@yamsergey Sergey did you have any script to automate the Momentics installation given it's interactive and asks to accept the EULA?

**Comment by yamsergey** [comment]
@vaskas  yes. I've managed to install it. So you can build your docker image. I will upload it to Docker hub soon (1-3 days). But so far you can check https://github.com/yamsergey/playbook-dev-tools/blob/master/.devcontainer/Dockerfile . It's my fork of initial play-book-dev. You can build container with `docker build -t {custom container tag}:{version e.g. 01} .devcontainer/` and then run it with mounted packages folder `docker run -it -v "$(pwd)":/root/packages {custom container tag}:{version e.g. 01} /bin/bash` then inside the container navigate to `/root/packages` and run `./build.sh -b /root/bbndk`  to build all packages. For me it usually looks like:
```
docker build -t bb10-dev-build:0.1 .devcontainer/
docker run -it -v "$(pwd):/root/packages" bb10-dev-build:0.1 /bin/bash
```

P.S. as a bonus you can run Momentics IDE if configure X11 server on your machine and pass correct `DISPLAY` parameter to the docker container. I've tested it on Mac OS and Windows with WSL2, finally can develop for my BB10 😂 all works quite good.

I've been trying to build this repo with the docker bvut still have few issues. But with original one it works.  Will keep you updated and create PR once I get all works.

**Comment by vaskas** [comment]
@yamsergey I tried the Dockerfile and hell it works great! Nice work! The only changes I made in the Dockerfile is updated the SDK level to `10.3.1.995` just in case, and the working directory to `/root/berrymuch` for consistency. You can open the PR and we'll all start using it!

**Comment by yamsergey** [comment]
We also can use Ubuntu 64Bit with additional libs from 32 bit, but I would prefer 32 bit images just for consistency.

**Comment by yamsergey** [comment]
I’m thinking about CI integration with CircleCI to keep the project health. But the whole build takes quite a long time. Do we have any adidas how to build per package? Like if I work on vim I want to build only VIM and when I finish the job, open PR and merge it we probably don’t want CI to spend another hour to build everything

**Comment by vaskas** [comment]
@yamsergey as a step 1 I would just trigger the whole build when something is merged into master. It would be also awesome to archive the produced packages somewhere.

As for the PRs I agree it would be great to only build the relevant packages. Maybe do it on the `git diff HEAD..master | grep ports` to figure out the affected ones?

**Comment by yamsergey** [comment]
@vaskas @berryamin Who can add user `berrymuchosci` to this repo?

**Comment by berryamin** [comment]
@yamsergey Wonderfull Sergey, thank you so much ! setting up the stuff right now ...
@berrymuchosci invitation link sent :)

**Comment by yamsergey** [comment]
Great! But unfortunately, the user has to have Admin access :(

**Comment by berryamin** [comment]
Not sure I'm able to give more than "write access to the repository" .. 
In the "Collaborators" part of the berrymuch project, there is no option at all. Only add collaborator ..

**Comment by berryamin** [comment]
Maybe we have to switch to an "Organisation" for the berrymuch project, as github describe it here
https://help.github.com/en/github/setting-up-and-managing-your-github-user-account/permission-levels-for-a-user-account-repository
As far as I understand, here are the rules that apply today
https://help.github.com/en/github/setting-up-and-managing-your-github-user-account/permission-levels-for-a-user-account-repository

**Comment by berryamin** [comment]
Ok I think this is the good way, let's go to an organisation. Are you able to create an organisation ? once it's up I'll transfer the project to it.

**Comment by berryamin** [comment]
Adding @vaskas just in case :)

**Comment by yamsergey** [comment]
Hmmm..I see. I will check about the organisation later. For now, will try to configure github actions instead of circleci

**Comment by berryamin** [comment]
When trying to build the docker image, I hit the following issue.

berryamin@shinwey ~ % docker build -t bb10-dev-build:0.1 .devcontainer/
unable to prepare context: path ".devcontainer/" not found

Sorry for what is probably a very basic docker question ..

But how may I create a .devcontainer/ ?

**Comment by vaskas** [comment]
Let's wait until Sergey submits the pull request for the dev container. Then we can grill him with questions and code review :smile:  I was also wondering why it needed .devcontainer and can't just have Dockerfile in the root dir.

On a different note, I am all for creating the organisation. I'll help consolidate all the other works and forks we are doing. Unfortunately https://github.com/berrymuch is already taken (`berrymuch-os` is not though)

**Comment by yamsergey** [comment]
Sorry for waiting. I'm planning to pen a PR with all explanations today :)

**Comment by berryamin** [comment]
Do you really want to keep the -os part ? we compile binaries, not an os, is it ? binberry was as perfect as berrymuch but already taken :)

**Comment by berryamin** [comment]
We're now hosted via the BerryFarm organisation :)

**Comment by vaskas** [comment]
Solved in #23
